logrusr.go

```diff
@@ -61,23 +95,44 @@ func (l *logrusr) Enabled() bool {
 	// logrus.InfoLevel has value 4 so if the level on the logger is set to 0 we
 	// should only be seen as enabled if the logrus logger has a severity of
 	// info or higher.
-	return int(log.GetLevel())-logrusDiffToInfo >= l.level
+	return log.IsLevelEnabled(logrus.Level(level + logrusDiffToInfo))
 }
 
-// V is a part of the Logger interface. Calling the method will change the
-// global log severity for the logr implementation.
-func (l *logrusr) V(level int) logr.InfoLogger {
-	newLogger := l.copyLogger()
-	newLogger.level = level
+// Info logs info messages if the logger is enabled, that is if the level on the
+// logger is set to logrus.InfoLevel or less.
+func (l *logrusr) Info(level int, msg string, keysAndValues ...interface{}) {
+	if !l.Enabled(level) {
```

Comment:
You don't need to check `Enabled()` - logr itself handles that and only calls you if true.